3scale/porta
# [WIP] Replaces Select2 with a custom searchable dropdown

**What this PR does / why we need it**:

- Removes Select2 from _Create Application_ form.
- Implements progressive enhancement with Formtastic
- Implements a `SearchableSelect` component with React that replaces form's vanilla selects.

**Which issue(s) this PR fixes** 

[THREESCALE-1585 Replace Select2 lib by React component](https://issues.jboss.org/browse/THREESCALE-1585)

**Verification steps**

1. Navigate to the new application page
2. Click in application/service plan: a dropdown with all plans should be displayed
3. Write something: plans should be filtered by name
4. Selecting an application plan should re-fresh _Service plans_ select with the proper options
5. Select any plan and submit: application should be created as usual
6. Disable Javascript (from browser dev tools) and verify all forms are available and an application can be created

**`SearchableSelect` in action**

![datalist](https://user-images.githubusercontent.com/11672286/49799368-1152a600-fd45-11e8-8e81-affcc2d61aad.gif)

Forms without Javascript (progressive enhancement):
<img width="740" alt="screen shot 2018-12-11 at 11 22 06" src="https://user-images.githubusercontent.com/11672286/49799403-24657600-fd45-11e8-894f-1c25b7d04cde.png">
